workflow/rules/enrichment.smk

```diff
@@ -0,0 +1,131 @@
+            category="Gene set enrichment analysis"
+            ),
+        significant=report(
+            "analysis/tables/fgsea/{model}.sig-gene-sets.tsv",
+            caption="../report/fgsea-table-significant.rst",
+            category="Gene set enrichment analysis"
+            ),
+        plot=report(
+            "analysis/plots/fgsea/{model}.table-plot.pdf",
+            caption="../report/fgsea-table-plot.rst",
+            category="Gene set enrichment analysis"
+            )
+    params:
+        species=config["resources"]["ref"]["species"],
+        model=get_model,
+        gene_set_fdr=config["enrichment"]["fgsea"]["fdr_gene_set"],
+        nperm=config["enrichment"]["fgsea"]["nperm"],
+        covariate=lambda w: config["diffexp"]["models"][w.model]["primary_variable"]
+    conda:
+        "../envs/fgsea.yaml"
+    threads: 8
+    script:
+        "../scripts/fgsea.R"
+
+rule fgsea_plot_gene_set:
+    input:
+        samples="analysis/sleuth/samples.tsv",
+        diffexp="analysis/tables/diffexp/{model}.genes-mostsigtrans.diffexp.tsv",
+        gene_sets=config["enrichment"]["gene_sets_file"],
+        sig_gene_sets="analysis/tables/fgsea/{model}.sig-gene-sets.tsv"
+    output:
+        report(
+            "analysis/plots/fgsea/{model}.{gene_set}.gene-set-plot.pdf",
+            caption="../report/fgsea-gene-set-plot.rst",
+            category="Gene set enrichment analysis"
+            )
+    params:
+        model=get_model,
+        covariate=lambda w: config["diffexp"]["models"][w.model]["primary_variable"]
+    conda:
+        "../envs/fgsea.yaml"
+    script:
+        "../scripts/fgsea_plot_gene_set.R"
+
+## gene ontology term enrichment analysis
+
+rule biomart_ens_gene_to_go:
+    output:
+        config["resources"]["ontology"]["ens_gene_to_go_file"]
+    params:
+        species=config["resources"]["ref"]["species"]
+    conda:
+        "../envs/biomart-download.yaml"
+    script:
+        "../scripts/biomart-ens_gene_to_go.R"
+
+
+rule download_go_obo:
+    output:
+        config["resources"]["ontology"]["gene_ontology_file"]
+    params:
+        download=config["resources"]["ontology"]["gene_ontology"]
+    conda:
+        "../envs/curl.yaml"
+    log:
+        "analysis/logs/curl/download_go_obo.log"
+    shell:
+        "( curl --silent -o {output} {params.download} ) 2> {log}"
+
+rule goatools_go_enrichment:
+    input:
+        obo=config["resources"]["ontology"]["gene_ontology_file"],
+        ens_gene_to_go=config["resources"]["ontology"]["ens_gene_to_go_file"],
+        diffexp="analysis/tables/diffexp/{model}.genes-mostsigtrans.diffexp.tsv"
+    output:
+        enrichment=report(
+            "analysis/tables/go_terms/{model}.genes-mostsigtrans.diffexp.go_term_enrichment.gene_fdr_{gene_fdr}.go_term_fdr_{go_term_fdr}.tsv",
+            caption="../report/go-enrichment-mostsigtrans-table.rst",
+            category="GO term enrichment analysis"
+            ),
```

Comment:
Please move all such trailing parentheses one indentation level to the left.

Reply:
https://github.com/snakemake-workflows/rna-seq-kallisto-sleuth/pull/4/commits/662d54762512fa194a40a83543662e7c519e1928